polkadot-parachain/src/command.rs

```diff
@@ -55,22 +56,30 @@ enum Runtime {
 	CollectivesWestend,
 }
 
-trait ChainType {
+trait RuntimeResolver {
 	fn runtime(&self) -> Runtime;
 }
 
-impl ChainType for dyn ChainSpec {
+impl RuntimeResolver for dyn ChainSpec {
 	fn runtime(&self) -> Runtime {
 		runtime(self.id())
 	}
 }
 
-use sc_chain_spec::GenericChainSpec;
-impl ChainType
-	for GenericChainSpec<rococo_parachain_runtime::GenesisConfig, chain_spec::Extensions>
-{
+/// Implementation, that can resolve [`Runtime`] from any json configuration file
+impl RuntimeResolver for PathBuf {
 	fn runtime(&self) -> Runtime {
-		runtime(self.id())
+		#[derive(Debug, serde::Deserialize)]
+		struct EmptyChainSpecWithId {
+			id: String,
+		}
+
+		let file = std::fs::File::open(self).expect("Failed to open file");
+		let reader = std::io::BufReader::new(file);
+		let chain_spec: EmptyChainSpecWithId = sp_serializer::from_reader(reader)
```

Comment:
How we obtained the id before, why we got one more file read and parse?

Reply:
old/actual approach, reads configuration path to GenericChainSpec based on testing/rococo (Genesis + Extentions), even if you have Statemint/Westmint/Whatever configuration file, e.g. once someone add new configuration which will use different Extentions (or maybe Geneis, not sure), it wont work, because serde will fail (unknown/missing attributes), and all of this is just because we need to know ID from the json file.

I was checking a possibilities to read chain_spec ignoring (genesis/extensions) but I did not find one.
Yess, now it does one more reading file, just ID from json. And then decides, what specific configuration to parse with anohter reading the file, so it wont fail on serde, so now it is more generic approach.

Reply:
Thank you for explaining. 
Make sense.